File: 2017000000040-Saleh-AdvancedJava-FinalProject-CodeBase/2017000000040-Saleh-Frontend-Adv-Java-Final-Project/object-models/BlockedUser.php
<?php

    class BlockedUser{
        public $id;
        public $dateBlocked;
        public $blockedUser;
        public $blockerUser;

        public function __construct($id, $dateBlocked, $blockedUser, $blockerUser){
            $this->id=$id;
            $this->dateBlocked=$dateBlocked;
            $this->blockedUser=$blockedUser;
            $this->blockerUser=$blockerUser;
        }
    }

?>

File: 2017000000040-Saleh-AdvancedJava-FinalProject-CodeBase/2017000000040-Saleh-Backend-Adv-Java-Final-Project/src/main/java/com/example/demo/model/BlockedUser.java
package com.example.demo.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.OneToOne;
import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class BlockedUser {
    @Id
    private Integer id;
    private LocalDate dateBlocked;
    @OneToOne
    private User blockedUser;
    @OneToOne
    private User blockerUser;
}
